web/src/desktop_notifications.ts

```diff
@@ -0,0 +1,91 @@
+import $ from "jquery";
+import assert from "minimalistic-assert";
+
+import type {Message} from "./message_store";
+
+// TODO(evy) -- can `obj` be ElectronBridgeNotification? if so, close() isn't defined
```

Comment:
the `obj` here is set in `message_notifications.js`:

```js
        notification_object = new desktop_notifications.NotificationAPI(title, {
            icon: icon_url,
            body: content,
            tag: message.id,
        });
        desktop_notifications.notice_memory.set(key, {
            obj: notification_object,
            msg_count,
            message_id: message.id,
        });
```

So the object is just a `NotificationAPI`, which should be either a `ElectronBridgeNotification` or `Notification`. It seems like in practice it's always a `NotificationAPI`? Or at least that `initialize` and `close_notification` are only called when it's a `NotificationAPI`, because `close` isn't defined on `ElectronBridgeNotification`.

I also have very limited understanding of Electron apps and what's going on there, so I wouldn't be surprised if I'm just not understanding how things work here. Would love some insight into what to do here :)

Reply:
`close` is defined on `ElectronBridgeNotification`, via the `Object.assign` statement in the constructor. TypeScript doesn’t understand the effect of `Object.assign`.

Reply:
Ah I see. So how do I handle the `close()` calls in this file?

Reply:
I don’t think there’s a better option than exploding the `Object.assign` statement into individual assignments that TypeScript understands.